File: backend/src/main/java/com/libbi/trivia/team/TeamService.java
package com.libbi.trivia.team;

import java.util.List;

public interface TeamService {
	
	TeamResponseDto createTeam(TeamRequestDto teamRequestDto, Long gameId);

	TeamResponseDto joinGame(Long teamId, Long gameId);

	TeamResponseDto changeTeamName(TeamRequestDto teamRequestDto, Long teamid);

	TeamResponseDto getUserTeam(Integer userId);
	
	Team getTeamById(Long teamId);

	TeamResponseDto createTeamLoggedIn(TeamRequestDto teamRequestDto);

	List<TeamResponseDto> getAllGameTeams(Long gameId);

	TeamResponseDto getTeam(Long teamId);

}

File: frontend/triviav2/src/app/team.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { BehaviorSubject, Observable } from 'rxjs';
import { TeamRequestDto } from "./models/TeamRequestDto";
import { TeamResponseDto } from "./models/TeamResponseDto";

const baseUrl = 'http://localhost:8080/api/v1/team';

@Injectable({
  providedIn: 'root'
})
export class TeamService {
  private teamNameSource = new BehaviorSubject<string>('');
  currentTeamName = this.teamNameSource.asObservable();
  constructor(private http: HttpClient) { }

  createTeam(teamRequestDto: TeamRequestDto, gameId: number): Observable<TeamResponseDto> {
    const payload = { ...teamRequestDto, gameId };
    return this.http.post<TeamResponseDto>(`${baseUrl}/${gameId}/new`, payload);
  }

  changeTeamName(teamRequestDto: TeamRequestDto, teamId: number): Observable<TeamResponseDto> {
    const payload = { ...teamRequestDto, teamId};
    return this.http.patch<TeamResponseDto>(`${baseUrl}/${teamId}/changename`, payload);
  }

  joinGame(teamId: number, gameId: number): Observable<TeamResponseDto> {
    return this.http.patch<TeamResponseDto>(`${baseUrl}/${teamId}/${gameId}/join`, {});
  }

  getTeamByUser(userId: number): Observable<TeamResponseDto> {
    return this.http.get<TeamResponseDto>(`${baseUrl}/${userId}/team`);
  }

  getAllGameTeams(gameId: number): Observable<TeamResponseDto[]> {
    return this.http.get<TeamResponseDto[]>(`${baseUrl}/${gameId}/teams`);
  }

  setTeamName(name: string) {
    this.teamNameSource.next(name);
  }
}
